website/docs/pipelines/promoting-applications.mdx

```diff
@@ -187,16 +206,108 @@ type: Opaque
+
+
+### Security Recommendations
+
+:::tip
+
+Adopt as much of the recommendations in this section to reduce the risks associated with the secrets involved in pull requests.
+
+:::
+
+1. **Create a user account for pull request changes**: this user context would be used to do any git provider operation,
+and for security and auditing perspective, you don't want to impersonate a real user for it.
+
+<details><summary>Expand to see example</summary>
+
+![using bot account](img/bot-account.png)
+</details>
+
+2. **Restrict access to the secret**: the promotion credentials needs to reside in the same Namespace as the Pipeline resource on the management cluster. Restrict
+via RBAC that only `pipeline-controller` service account is able to read it.
+
+<details><summary>Expand to see example</summary>
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: read-app-promotion-credentials
+  namespace: app
+rules:
+  - apiGroups:
+      - ""
+    resourceNames:
+      - "app-promotion-credentials"
+    resources:
+      - "secrets"
+    verbs:
+      - "get"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: pipeline-controller-read-app-promotion-credentials
+  namespace: app
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: read-app-promotion-credentials
+subjects:
+  - kind: ServiceAccount
+    name: pipeline-controller
+    namespace: pipeline-system
+```
+</details>
+
+3. **Do not use long-live tokens**: set an expiration date and rotate them according to your security policy.
+
+<details><summary>Expand to see example</summary>
+
+![create token with expiration](img/create-token-with-expiration.png)
+</details>
+
+4. **Honour the least privilege principle**: avoid having high privilege tokens. Restrict the token to your just your repo and to just the operations required.
+
+<details><summary>Expand to see example</summary>
+
+For example, if the case of GitHub, use [fine-grained tokens](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/) to only
+allow access to the single repo that your configuration manifests exist.
+
+![create token least priviledge](img/fine-grained-token.png)
+ </details>
+
+5. **Review active access tokens on a regular basis**: to ensure that only the ones that are required are present at all times.
+
+<details><summary>Expand to see example</summary>
+
+For example, using github and fine-grained tokens you [could do so](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/#approving-and-auditing-personal-access-tokens).
+
+![review tokens](img/manage-fine-grained.png)
+</details>
+
+6. **Reduce number of secrets to maintain**: Whenever ends in a more secure setup, favour git over https with tokens instead of ssh and deploy keys. Given that a token
```

Comment:
Can we rephrase the first part of the sentence `Whenever ends ..` to make it more clear?

Reply:
reworded, let me know if its a bit more clear